ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -136,7 +152,32 @@ ompi_mtl_ofi_component_register(void)
                                     MCA_BASE_VAR_SCOPE_READONLY,
                                     &ompi_mtl_ofi.ofi_progress_event_count);
 
-     free(desc);
+    free(desc);
+
+    ret = mca_base_var_enum_create ("ofi_tag_mode_type", ofi_tag_mode_type , &new_enum);
+    if (OPAL_SUCCESS != ret) {
+        return ret;
+    }
+
+    ofi_tag_mode = MTL_OFI_TAG_AUTO;
+    asprintf(&desc, " Mode for OFI tag."
+            " 1 auto (default): detect if the provider supports FI_REMOTE_CQ_DATA or fallback to 2."
+            " 2 ofi_tag_1: %d bits com_id,%d bits source rank,%d bits mpi_tag."
+            " 3 ofi_tag_2: %d bits com_id,%d bits source rank,%d bits mpi_tag."
+            " 4 force_fi_cq_data: try FI_REMOTE_CQ_DATA or fail if not supported.",
```

Comment:
Is the user supposed to specify a value of `auto`, `ofi_tag_1`, ...etc., or a value of `1`, `2`, ...etc.?  From reading this help message, I think a user might assume that they are supposed to use the number, but I think that since you're registering this as an MCA var enum, they're supposed to specify the names (`auto`, etc.), right?

Here's my suggestion for the help message:

> Mode specifying how many bits to use for various MPI values in OFI/Libfabric communications.  Some Libfabric provider network types can support as many bits as Open MPI needs; others can only supply a limited number of bits, which then must be split across the MPI communicator ID, MPI source rank, and MPI tag.  Three different splitting schemes are available: ofi_tag_full (%d bits for the communicator, %d bits for the source rank, and %d bits for the tag), ofi_tag_2 (%d bits communicator, %d bits source rank, %d bits tag), or ofi_tag_3 (%d bits communicator, %d bits source rank, %d bits tag).  By default, this MCA variable is set to "auto", which will first try to use "ofi_tag_full", and if that fails, fall back to "ofi_tag_2".

Where "ofi_tag_full" -- I think -- is the equivalent to your force_fi_cq_data (FI_REMOTE_CQ_DATA and friends have no meaning to the end user).

Reply:
passing numbers vs strings: both will work (since the enum is registered), but (please correct me if I'm wrong) most of the mca parameters are written suggesting text. I take it in and will update the README accordingly and also update the parameter help message. Thanks

Reply:
Cool, thanks.  I had honestly forgotten that we allow both numbers or names for enum values of MCA vars.  If you find my suggested text helpful, awesome.  😄